### PR TITLE
product rating

### PR DESCRIPTION
**Pull request by nhungltm**
screen shot 2018-08-23 at 21 32 06
https://edu-redmine.framgia.vn/issues/744

### CODE REVIEW

**Comment by nhungltm** [comment]
ready

File: app/models/average_cache.rb [hunk 1]
```diff
@@ -0,0 +1,4 @@
+class AverageCache < ApplicationRecord
+  belongs_to :rater, class_name: "User"
```
Comment (huuphuc93) [inline app/models/average_cache.rb]: User.name

File: app/views/products/_product-detail.html.erb [hunk 2]
```diff
@@ -34,7 +34,13 @@
     <div class="product-information">
       <%= image_tag "product-details/new.jpg", class: "newarrival" %>
       <h2><%= @product.name %></h2>
-      <%= image_tag "product-details/rating.png" %>
+      <p>
+        <b>
+          <%= rating_for @product, "quality", disable_after_rate: true, cancel: false, star: Settings.num_stars %>
```
Comment (huuphuc93) [inline app/views/products/_product-detail.html.erb]: "quality" => :quality

**Comment by huuphuc93** [comment]
commented
comment nhỏ

**Comment by nhungltm** [comment]
ready

File: app/models/rate.rb [hunk 3]
```diff
@@ -0,0 +1,4 @@
+class Rate < ApplicationRecord
+  belongs_to :rater, class_name: "User"
```
Comment (huuphuc93) [inline app/models/rate.rb]: "User" => User.name

**Comment by huuphuc93** [comment]
commented
comment nhỏ

**Comment by nhungltm** [comment]
ready

**Comment by huuphuc93** [comment]
merged
this (pr) can be (merged) now
CC: [To:2076564] Nguyen Huy Hung C (93)
CC: [To:1818111] Nguyen Thi Huong (94)
CC: [To:1546746] Md. Mahmudur Rahman (Mahmud)

**Pull request merged**